### PR TITLE
Add Maven

### PR DESCRIPTION
**Pull request by Tobias321**
see #58 

run this with:
`mvn clean compile`
and
`mvn exec:java -Dexec.mainClass="de.pmdungeon.desktop.Main"`

### CODE REVIEW

**Review by AMatutat — APPROVED** [review]
Works for me. 

Should we add the maven commands in the readme?

**Comment by Tobias321** [comment]
> Works for me.

works for me too.

> Should we add the maven commands in the readme?

yes 🙂

**Review by cagix — APPROVED** [review]
lgtm 👍 

just did a formal review, did not test-run the code ...

**Comment by AMatutat** [comment]
i would suggest 

"First run `mvn clean compile` in your project folder than use `mvn exec:java -Dexec.mainClass="de.pmdungeon.PACKAGE.MAIN"` to start the game. "

for the readme.

**Comment by cagix** [comment]
> i would suggest
> 
> "First run `mvn clean compile` in your project folder than use `mvn exec:java -Dexec.mainClass="de.pmdungeon.PACKAGE.MAIN"` to start the game. "

true that. but on the other hand, shouldn't maven be so clever as to figure out to compile things first before starting things (if needed)? at least in ant and gradle you can configure such dependencies ...

**Comment by Tobias321** [comment]
Sounds good 😊

**Comment by Tobias321** [comment]
@cagix Afaik, you can define additional goals in the pom.xml.

**Comment by AMatutat** [comment]
> true that. but on the other hand, shouldn't maven be so clever as to figure out to compile things first before starting things (if needed)? at least in ant and gradle you can configure such dependencies ...

The question is: What is our target group for this project?  I actually want to keep the entry hurdle as low as possible.

File: README.md [hunk 1]
```diff
@@ -26,8 +26,10 @@ TODO
 
 ### ANT
 - TODO
+
 ### Maven
-- TODO
+- First run `mvn clean compile` in your project folder than use `mvn exec:java -Dexec.mainClass="de.pmdungeon.PACKAGE.MAIN"` to start the game.
```
Comment (cagix) [inline README.md:31]: "project folder than use" => "project folder, then use"

**Review by cagix — CHANGES_REQUESTED** [review]
small typo

File: README.md (hunk 1 above)
Comment (Tobias321) [reply README.md]: fixed

**Comment by cagix** [comment]
> > true that. but on the other hand, shouldn't maven be so clever as to figure out to compile things first before starting things (if needed)? at least in ant and gradle you can configure such dependencies ...
> 
> The question is: What is our target group for this project? I actually want to keep the entry hurdle as low as possible.

i tend to agree :)  ... however, it could be easier if the scripts behave consistently. With Gradle and Ant we have implemented appropriate dependencies, i.e. you can directly call "run" and it will be compiled in the background first (if necessary). i can't think of any reason why we should do this differently in Maven?

File: README.md (hunk 1 above)
Comment (cagix) [reply README.md]: also, why "`clean compile`"? wouldn't "`compile`" suffice?

**Comment by Tobias321** [comment]
`mvn clean compile exec:java` should work too

main class can defined here:
```
<properties>
        <exec.mainClass>fully-qualified-class-name</exec.mainClass>
</properties>
```

**Comment by cagix** [comment]
> `mvn clean compile exec:java` should work too

yup, my complaint is this: why isn't just `mvn exec:java` enough to compile and start?

File: README.md (hunk 1 above)
Comment (Tobias321) [reply README.md]: only `compile` should be ok

**Comment by Tobias321** [comment]
> > `mvn clean compile exec:java` should work too
> 
> yup, my complaint is this: why isn't just `mvn exec:java` enough to compile and start?

i understand. think, we need the org.codehaus.mojo exec-maven-plugin

**Review by cagix — APPROVED** [review]

**Review by cagix — APPROVED** [review]

**Pull request merged**